New version adapted to nullable columns and null handling in multi-stage engine (#304)
pinot-contrib/pinot-docs: configuration-reference/schema.md

#   Schema

Schema is used to define the names, data types, and other information for the columns of a Pinot table.

The Pinot schema is composed of:

> | Field                              | Since | Default   | Description                                                                                                                                                                                                                                           |

Reviewer: is the release version included the Pinot schema? 
if yes, perhaps we can reword Since to Release version? 
we try not to use the term [since](https://developers.google.com/style/word-list#since)

Author: Changed

|------------------------------------|-------|-----------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | **schemaName**                     | -     | mandatory | Defines the name of the schema. This is usually the same as the table name. The offline and the real-time table of a hybrid table should use the same schema.                                                                                         |

Reviewer: ```suggestion
| **schemaName**                     | -     | Required | Name of the schema. Usually the same as the table name. The offline and the real-time table of a hybrid table should use the same schema.                                                                                         |
```

Reviewer: by usually, do we mean this happens by default?

Author: Changed. New line is 

```
| **schemaName**                     | -               | required | Name of the schema. This must be the same as the table name without the REALTIME or OFFLINE suffix. Therefore, the offline and the real-time table of a hybrid table should use the same schema.                              |
```

> | **schemaName**                     | -     | mandatory | Defines the name of the schema. This is usually the same as the table name. The offline and the real-time table of a hybrid table should use the same schema.                                                                                         |

Reviewer 2: Schema name must be the raw table name. We enforced that in 1.1.0

Author: Changed

> | **enableColumnBasedNullHandling**  | 1.1.0 | false     | Whether to enable column-based or table-based null handling. The default value `false` means that Pinot will use a table-based null handling. See [Null value support](../developers/advanced/null-value-support.md) for more information about this. |

Reviewer: ```suggestion
| **enableColumnBasedNullHandling**  | 1.1.0 | false     | Enable column-based or table-based null handling. By default `false` means Pinot uses table-based null handling. For more information about this, see [Null value support](../developers/advanced/null-value-support.md). |
```

Reviewer: does setting this to`true` mean that Pinot uses column-based null handling?

Author: Yes, and it is explained in the referenced section

Author: The suggested:
> By default `false` means Pinot uses table-based null handling. For more information about this, see [Null value

Seems to indicate that the default behavior when false is set could be changed and that is not the case.

What about something like:
> When set to `true`, enables column-based null handling. The default value `false` means to use table-based null handling

Author: Changed to:

> When set to true, enables column-based null handling. The default value false means to use table-based null handling

> | **dimensionFieldSpec**             | -     | []        | A dimensionFieldSpec is defined for each dimension column. For more details, scroll down to [DimensionFieldSpec](schema.md#dimensionfieldspec).                                                                                                       |

Reviewer: ```suggestion
| **dimensionFieldSpec**             | -     | []        | A dimensionFieldSpec is defined for each dimension column. For more details, see [DimensionFieldSpec](schema.md#dimensionfieldspec).                                                                                                       |
```

> | **metricFieldSpec**                | -     | []        | A metricFieldSpec is defined for each metric column. For more details, scroll down to [MetricFieldSpec](schema.md#metricfieldspec).                                                                                                                   |

Reviewer: ```suggestion
| **metricFieldSpec**                | -     | []        | A metricFieldSpec is defined for each metric column. For more details, see [MetricFieldSpec](schema.md#metricfieldspec).                                                                                                                   |
```

> | **dateTimeFieldSpec**              | -     | []        | A dateTimeFieldSpec is defined for the time columns. There can be multiple time columns. For more details, scroll down to [DateTimeFieldSpec](schema.md#datetimefieldspec).                                                                           |

Reviewer: ```suggestion
| **dateTimeFieldSpec**              | -     | []        | A dateTimeFieldSpec is defined for the time columns. There can be multiple time columns. For more details, see [DateTimeFieldSpec](schema.md#datetimefieldspec).                                                                           |
```
